### ISSUE TITLE
Failed to fetch models from remote server

### ISSUE THREAD

**Issue by idhamari** [Project-MONAI/MONAILabel#297]
**Describe the bug**
Failed to fetch models from remote server. Make sure server address is correct and <server_uri>/info/ is accessible in browser

**To Reproduce**
* I downloaded the plugin and add it to slicer path.
*   I run these commands: 

```
   pip3 install monailabel
  # download sample apps/dataset
  monailabel apps --download --name deepedit_left_atrium --output apps
  monailabel datasets --download --name Task02_Heart --output datasets
  
  # run server
  monailabel start_server --app apps\deepedit_left_atrium --studies datasets\Task02_Heart\image

```

* I open the module in slicer then add **http://0.0.0.0:8000/** to **MONAI Label server** field then click **Fetch/Rferech Models from server** button

**Expected behavior**
Don't know, I just start testing this plugin. I guess it should load some models.  

**Environment**

ubuntu, pyhton 3.6,  Slicer 4.11, and  tf2

```
python3 -c 'import monai; monai.config.print_debug_info()'
================================
Printing MONAI config...
================================
MONAI version: 0.6.0
Numpy version: 1.19.5
Pytorch version: 1.8.1+cu102
MONAI flags: HAS_EXT = False, USE_COMPILED = False
MONAI rev id: 0ad9e73639e30f4f1af5a1f4a45da9cb09930179

Optional dependencies:
Pytorch Ignite version: 0.4.5
Nibabel version: 3.2.1
scikit-image version: 0.17.2
Pillow version: 8.2.0
Tensorboard version: 2.0.2
gdown version: 3.13.0
TorchVision version: 0.9.1+cu102
ITK version: 5.1.2
tqdm version: 4.61.1
lmdb version: 1.2.1
psutil version: 5.8.0
pandas version: 1.1.5
einops version: NOT INSTALLED or UNKNOWN VERSION.

For details about installing the optional dependencies, please visit:
    https://docs.monai.io/en/latest/installation.html#installing-the-recommended-dependencies


================================
Printing system config...
================================
System: Linux
Linux version: Ubuntu 18.04.5 LTS
Platform: Linux-5.4.0-81-generic-x86_64-with-Ubuntu-18.04-bionic
Processor: x86_64
Machine: x86_64
Python version: 3.6.9
Process name: python3
Command: ['python3', '-c', 'import monai; monai.config.print_debug_info()']
Open files: []
Num physical CPUs: 12
Num logical CPUs: 24
Num usable CPUs: 24
CPU usage (%): [11.8, 11.8, 92.8, 10.3, 10.4, 10.6, 15.2, 10.4, 1.5, 14.9, 16.4, 14.5, 11.8, 11.8, 11.8, 10.4, 13.0, 11.8, 11.8, 11.6, 10.3, 11.8, 16.7, 19.1]
CPU freq. (MHz): 2315
Load avg. in last 1, 5, 15 mins (%): [2.6, 2.8, 2.6]
Disk usage (%): 48.7
Avg. sensor temp. (Celsius): UNKNOWN for given OS
Total physical memory (GB): 62.8
Available memory (GB): 46.8
Used memory (GB): 7.2

================================
Printing GPU config...
================================
Num GPUs: 2
Has CUDA: True
CUDA version: 10.2
cuDNN enabled: True
cuDNN version: 7605
Current device: 0
Library compiled for CUDA architectures: ['sm_37', 'sm_50', 'sm_60', 'sm_70']
GPU 0 Name: NVIDIA GeForce RTX 2080 Ti
GPU 0 Is integrated: False
GPU 0 Is multi GPU board: False
GPU 0 Multi processor count: 68
GPU 0 Total memory (GB): 10.8
GPU 0 CUDA capability (maj.min): 7.5
GPU 1 Name: NVIDIA GeForce RTX 2080 Ti
GPU 1 Is integrated: False
GPU 1 Is multi GPU board: False
GPU 1 Multi processor count: 68
GPU 1 Total memory (GB): 10.8
GPU 1 CUDA capability (maj.min): 7.5
```

**Additional context**

the link http://0.0.0.0:8000/info/ works in the browser with htis output

{"version":1,"name":"DeepEdit - Left Atrium","description":"MONAI Label App to provide active learning solution using DeepEdit to label left atrium over 3D MRI Images","dimension":3,"config":{"infer":{"device":"cuda"},"train":{"name":"model_01","pretrained":false,"device":"cuda","amp":true,"lr":0.0001,"epochs":50,"val_split":0.2,"train_batch_size":1,"val_batch_size":1}},"models":{"deepedit":{"type":"deepgrow","labels":[],"dimension":3,"description":"A pre-trained 3D DeepGrow model based on UNET"},"left_atrium":{"type":"segmentation","labels":["left atrium"],"dimension":3,"description":"A pre-trained model for volumetric (3D) segmentation of the left atrium over 3D MR Images"}},"strategies":{"random":{"description":"Random Strategy"},"first":{"description":"Get First Sample"}},"scoring":{"sum":{"description":"Compute Numpy Sum for Final/Original Labels"},"dice":{"description":"Compute Dice for predicated label vs submitted"}},"labels":["left atrium"],"train_stats":{},"datastore":{"total":20,"completed":0,"label_tags":{},"train":[]}}

**Comment by SachidanandAlle** [comment]
Try with empty url.. and let c if default works..

**Comment by idhamari** [comment]
same error!

**Comment by idhamari** [comment]
I redownload the version from this [commit](https://github.com/Project-MONAI/MONAILabel/pull/264) and it works now.

**Comment by idhamari** [comment]
Thanks for the info. I confirm that the last version work. 

I also found out the main reason for the error above. It seems if there is a space at the beginning, I get the same error above e.g.  " http://0.0.0.0:8000"

This could be fixed in the code by removing white spaces from both ends.

**Comment by SachidanandAlle** [comment]
> Thanks for the info. I confirm that the last version work.
> 
> I also found out the main reason for the error above. It seems if there is a space at the beginning, I get the same error above e.g. " http://0.0.0.0:8000"
> 
> This could be fixed in the code by removing white spaces from both ends.

thanks for reporting the spacing issue.. we can get that fixed :)

**Comment by Antoniaaa** [comment]
> Thanks for the info. I confirm that the last version work.
> 
> I also found out the main reason for the error above. It seems if there is a space at the beginning, I get the same error above e.g. " http://0.0.0.0:8000"
> 
> This could be fixed in the code by removing white spaces from both ends.

where to find the " http://0.0.0.0:8000/"

**Comment by diazandr3s** [comment]
> > Thanks for the info. I confirm that the last version work.
> > I also found out the main reason for the error above. It seems if there is a space at the beginning, I get the same error above e.g. " http://0.0.0.0:8000"
> > This could be fixed in the code by removing white spaces from both ends.
> 
> where to find the " http://0.0.0.0:8000/"

Hi @Antoniaaa,

"http://0.0.0.0:8000/" is the local IP address. Once you start the MONAI Label server you should be able to access it.

**Comment by MatthiasCami8** [comment]
I got the same issue on Linux as originally stated here. Leaving the url empty does not work either. The localhost:8000 path (and the localhost:8000/info) DO work in my browser, so the server has started successfully but Slicer does not seem to find it. I tried the current stable release (5.2.2) and the preview release (5.3.0) but both don't work.

**Comment by diazandr3s** [comment]
Are the server and Slicer running on the same PC? If yes, Slicer needs the "http://" as well. Please try http://localhost:8000

**Comment by MatthiasCami8** [comment]
They are not. For full information: the server is running on a VM on GCP. But after setup everything seems to work fine there, and as I said I'm able to access the server from my local browser (after tunneling to my remote pc). However Slicer does not seem to find it.
